nodejs/src/session_options_helper.cc

```diff
@@ -0,0 +1,144 @@
+#include "onnxruntime_cxx_api.h"
+#include <napi.h>
+
+#include <unordered_map>
+
+#include "common.h"
+#include "session_options_helper.h"
+
+const std::unordered_map<std::string, GraphOptimizationLevel> GRAPH_OPT_LEVEL_NAME_TO_ID_MAP = {
+    {"disabled", ORT_DISABLE_ALL},
+    {"basic", ORT_ENABLE_BASIC},
+    {"extended", ORT_ENABLE_EXTENDED},
+    {"all", ORT_ENABLE_ALL}};
+
+const std::unordered_map<std::string, ExecutionMode> EXECUTION_MODE_NAME_TO_ID_MAP = {{"sequential", ORT_SEQUENTIAL},
+                                                                                      {"parallel", ORT_PARALLEL}};
+
+void ParseExecutionProviders(const Napi::Array epList, Ort::SessionOptions &sessionOptions) {
+  for (uint32_t i = 0; i < epList.Length(); i++) {
+    Napi::Value epValue = epList[i];
+    std::string name;
+    if (epValue.IsString()) {
+      name = epValue.As<Napi::String>().Utf8Value();
+    } else if (!epValue.IsObject() || epValue.IsNull() || !epValue.As<Napi::Object>().Has("name") ||
+               !epValue.As<Napi::Object>().Get("name").IsString()) {
+      ORT_NAPI_THROW_TYPEERROR(epList.Env(), "Invalid argument: sessionOptions.executionProviders[", i,
+                               "] must be either a string or an object with property 'name'.");
+    } else {
+      name = epValue.As<Napi::Object>().Get("name").As<Napi::String>().Utf8Value();
+    }
+
+    // CPU execution provider
+    if (name == "cpu") {
+      // TODO: handling CPU EP options
```

Comment:
We don't support handling CPU EP options for this release ?

Reply:
we don't yet support any execution provider **in session options**. one reason is: in this version the implementation only support CPU EP, so there is no point of support other EP in session options. the other reason is: the current C++ API does not have a interface to add EP in session options (which is a missing and should be added).